socketio/src/asynchronous/client/client.rs

```diff
@@ -375,9 +379,13 @@ impl Client {
                     }
                 }
                 PacketId::Connect => {
+                    let mut server_disconnected = self.server_disconnected.write().await;
+                    *server_disconnected = false;
```

Comment:
For what case is this necessary? Given that we initialize it with false?

Reply:
It gets set to `true` in the `PacketId::Disconnect` case. If we were configured to reconnect and it succeeds, this flag should be reset so it doesn't confuse us when the next disconnection happens (which may not be due to a server disconnect at all).

Reply:
Makes sense! Thanks for the explanation.